source/lz4.c

```diff
@@ -986,6 +986,60 @@ int LZ4_loadDict (LZ4_stream_t* LZ4_dict, const char* dictionary, int dictSize)
     return dict->dictSize;
 }
 
+int LZ4_stream_preserve(LZ4_stream_t* stream_, struct preserved_hash_table_entry** buf_out)
+{
+  LZ4_stream_t_internal* stream = (LZ4_stream_t_internal*)stream_;
+
+  int num_locs_needed = 0;
+  for (int i = 0; i < HASH_SIZE_U32; i++)
+  {
+    if (stream->hashTable[i] != 0)
+    {
+      num_locs_needed++;
+    }
+  }
+
+  // Add an extra location - this holds the sentinel value to indicate we're at the end of the array
+  num_locs_needed += 1;
+
+  int nbytes = sizeof(struct preserved_hash_table_entry) * num_locs_needed;
+  struct preserved_hash_table_entry* buf = malloc(nbytes);
+  memset(buf, -1, nbytes);
```

Comment:
Your sentinel is an entry with the location field set to -1. It is just a coincidence that memsetting an integer to -1 gives the same result as assigning -1 to it. Could you instead explicitly set the sentinel location to -1?